### DISCUSSION TITLE
How to find a commit hash that yields a specific version of OpenRoad build from source ?

### DISCUSSION THREAD

**Discussion by fawqati-spherical** [The-OpenROAD-Project/OpenROAD#5556] [Q&A] [answered]
Background: 
Since im on a Rocky8 OS, im having to build the binaries from source. My method is to :

1. git clone the repo
2. git revert to a commit hash
3. install the dependancies 
4. build the binaries with make/cmake

Please let me know if that isnt right.

My issue is that i've found our flow needs specific versions of the following combination: 
- python=3.11.9
- openlane=2.0.4
- openroad=v2.0-13269-g2c2f9599a

So the BIG QUESTION is how to find which commit hash yields a specific OpenRoad version ???

Your help is sincerely appreciated in advance.

Best,
Faisal

**Comment by maliberty** [comment]
Why can't you use the head of master and need to revert?

What do you mean by "OpenRoad version"?  We don't do formal releases so any given commit is a "version".

**Reply by fawqati-spherical** [reply]
Hi maliberty,
After the build i execute 
> openroad -version 
from the command line and i get different versions results for different commits. 

I know the OR version i need is = v2.0-13269-g2c2f9599a. 
What i dont know is which commit hash would yield that version ?!

**Reply by maliberty** [reply, marked as answer]
2c2f9599a  (everything after -g)

**Reply by fawqati-spherical** [reply]
Good morning maliberty,

You were right. All that was needed was a checkout without the 'g'

`git checkout 2c2f9599a
`
Thank you for your help. Much appreciated.
Faisal
